### DISCUSSION TITLE
Next.JS 13 Turbo - Error Boundaries Not Working

### DISCUSSION THREAD

**Discussion by tomitrescak** [vercel/next.js#42322] [Help]
Hello,

Before I post this to bugs, does anyone suffer with this probelm?
I have M1 Mac and I do the following:

```
pnpm create next-app --example with-turbopack
```

I run the example, but when tesing the Erro boundaries the whole application explodes and I get this in the console:

```
[404] /__nextjs_original-stack-frame (0.861ms)
[404] /__nextjs_original-stack-frame (0.373ms)
```

And this on the front end:

![image](https://user-images.githubusercontent.com/2682705/199366223-02cc5420-84a9-495e-bed5-2df0fb1fbd0a.png)

I thought that the whole purpose of the error boundary is to catch the error within app?

**Comment by icyJoseph** [comment]
## Edit

I did miss the `--turbo` part of your question... Consider that TurboPack is in alpha, if something looks odd, then it is good to test it against the regular flow first.

It does look like errors with `--turbo` go out of control and the entire tree falls apart...

---


Error boundaries do not catch SSR errors, and also during dev mode the error overlay surfaces above the app, often you have to close the error overlay to see if the UI is behaving as you expect, or build for production. 

Could you share a repro demo that we can talk over?

When using the boundary like:

```tsx
export default function Page() {
  return (
    <div>
      Landing page
      <ErrorBoundary>
        <Counter />
      </ErrorBoundary>
    </div>
  );
}
```

Where `Counter`:

```tsx
"use client";

import { useState } from "react";

export default function Counter() {
  console.log("Btn"); // keeping this here because I included it in the screenshots too
  const [count, setCount] = useState(0);

  if (count === 5) {
    throw new Error("Kaboom");
  }

  return (
    <button onClick={() => setCount((x) => x + 1)}>Count is: {count}</button>
  );
}
```

Dev mode:

![Screenshot 2022-11-02 at 09 30 38](https://user-images.githubusercontent.com/21013447/199438751-92be4ff9-a81c-4487-a134-0deceedcedf5.png)

Production:

![Screenshot 2022-11-02 at 09 33 28](https://user-images.githubusercontent.com/21013447/199439294-833812f6-7939-4512-ba90-f5f6f4bdd476.png)

Or perhaps you are using `error.tsx` pages?

Dev mode:

![Screenshot 2022-11-02 at 09 37 15](https://user-images.githubusercontent.com/21013447/199440093-9c1c1062-8e9d-4bd4-9669-759e4cade04a.png)

Prod mode with error.tsx page:

![Screenshot 2022-11-02 at 09 36 41](https://user-images.githubusercontent.com/21013447/199439983-e77a035e-3516-46f8-8bab-6b5b904c0d00.png)

Do notice that with `error.tsx`, that entire subtree is replaced by the error boundary (the landing page text is gone).
